openmc/material.py

```diff
@@ -249,15 +249,11 @@ def isotropic(self, isotropic):
 
     @property
     def fissionable_mass(self):
-        if self.volume is None:
-            raise ValueError("Volume must be set in order to determine mass.")
-        density = 0.0
-        for nuc, atoms_per_cc in self.get_nuclide_atom_densities().values():
-            Z = openmc.data.zam(nuc)[0]
-            if Z >= 90:
-                density += 1e24 * atoms_per_cc * openmc.data.atomic_mass(nuc) \
-                           / openmc.data.AVOGADRO
-        return density*self.volume
+        mass = 0.0
+        for nuc, _, _ in self._nuclides:
+            if openmc.data.zam(nuc)[0] >= 90:
+                mass += self.get_mass(nuc)
```

Comment:
Calling `get_mass(nuc)` involves a loop over each nuclide in the material. Since this call is already in a loop over the nuclides, that would make this algorithm O(N^2) to determine the fissionable mass, While the version you have is certainly more straightforward than the original, I do worry that if you have a lot of materials with a lot of nuclides, this could end up slowing things down.

Reply:
I wanted to remove the duplicate code, but you're right, the performance is a concern. I tested the performance: for one material with 200+nuclides, the version using `get_mass()` slows down by one order of magnitude, from around `1 ms` to `10+ ms` per function call. Reverted.